packages/gensx-storage/src/blob/provider.tsx

```diff
@@ -23,16 +26,33 @@ import { BlobProviderProps } from "./types.js";
  */
 export const BlobProvider = Component<BlobProviderProps, never>(
   "BlobProvider",
-  (props) => {
+  async (props) => {
+    const kind =
+      "kind" in props
+        ? props.kind
+        : process.env.GENSX_RUNTIME === "cloud"
+          ? "cloud"
+          : "filesystem";
+
+    const rootDir =
+      "rootDir" in props
+        ? props.rootDir!
+        : join(process.cwd(), ".gensx", "blobs");
+
     // Create the appropriate storage implementation based on kind
-    if (props.kind === "filesystem") {
-      const { rootDir = process.cwd(), defaultPrefix } = props;
-      const storage = new FileSystemBlobStorage(rootDir, defaultPrefix);
+    if (kind === "filesystem") {
+      // Ensure the storage directory exists
+      await mkdir(rootDir, { recursive: true });
+      const storage = new FileSystemBlobStorage(rootDir, props.defaultPrefix);
       return <BlobContext.Provider value={storage} />;
     } else {
       // Must be cloud based on our type definitions
-      const { defaultPrefix, organizationId } = props;
-      const storage = new RemoteBlobStorage(defaultPrefix, organizationId);
+      const organizationId =
```

Comment:
Is this not also read from an environment variable? 

Executor should probably set `GENSX_ORG` and we should be able to read this from the ondisk config locally (or the same env var if the user sets it).

Reply:
And is this actually an ID, or is it the organization name?

Reply:
Yeah i guess we should remove this, i'm not sure it's actually used. this was something that was in the original code that i didn't touch.

Reply:
removed; definitely a good improvement